### PR TITLE
[Main Course] - Docker sharing

### PR DESCRIPTION
**Pull request by Tina0330**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jhqian0909 — APPROVED** [review]

**Comment by jhqian0909** [comment]
Docker这篇放在甜点里是不是不太合适，感觉还是挺硬核的东西

**Review by Realvincentyuan — APPROVED** [review]
可以留点彩蛋提示下一次写什么，因为现在的信息感觉有些太浅显了

**Comment by Tina0330** [comment]
> 可以留点彩蛋提示下一次写什么，因为现在的信息感觉有些太浅显了

下次聊聊如何实现云服务器上的自动化部署，可以吗？

File: docs/Dessert/Tactics/2021-11-05-Docker-sharing.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+谈到Docker，自然离不开镜像(image)，容器（container），镜像仓库（repository）这三个概念。这里我将用一个贴近现实的比喻，帮助你更形象地理解这三个概念。
+
+试想我们的产品是一辆小汽车，如果有消费者欣赏这款车的车型，没有Docker他可能需要从零件开始复刻这辆车的制作过程。
+
+镜像(image)就好比这个汽车的雏形，是这台车刚生产出来的样子，只有轮胎，发动机，方向盘等基础配件。镜像在这里决定着一款车的汽车型号。
+
+容器，就相当于一款车型在市场中为了迎合不同口味的而设计出的不同版本，如豪华版，简易版等，甚至买家在购买之后根据自己的喜好装饰它。也就是说，容器是基于镜像的运用实列。容器和容器之间是相互独立的，但有可能是来源于同一个镜像。通过命令，我们可以创建，运行，停止和删除容器。
+
+镜像仓库，就像是停车场，存放着各种各样不同的车。这个比较好理解，这个仓库概念很像Github Repo,是存放所有镜像的地方。
```
[the remark this answers is not in the document]
Comment (Tina0330) [reply docs/Dessert/Tactics/2021-11-05-Docker-sharing.md]: 佳辉，改了。

**Comment by Realvincentyuan** [comment]
> > 可以留点彩蛋提示下一次写什么，因为现在的信息感觉有些太浅显了
> 
> 
> 
> 下次聊聊如何实现云服务器上的自动化部署，可以吗？

见微信，这个文章放在analytics tool folder里吧，Tina

File: docs/Main_Course/Data_Analytics/2021-11-05-Docker-Sharing.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+docker build -t /path/to/Dockerfile .
+```
+
+### 3.3 运行容器及常见命令
+容器就是运行起来的镜像，通过查看镜像选择你想要运行的镜像就可以启动容器：
+
+```Docker
+docker run <image_id>  #新建容器并启动
+```
+值得注意的是，`docker run` 有两个常见参数， `docker run -d` 表示后台运行， 而`docker run -p` 在是端口运行。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-11-05-Docker-Sharing.md:70]: 端口映射也应该提提

File: docs/Main_Course/Data_Analytics/2021-11-05-Docker-Sharing.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+最后，附上常用的基本操作可以帮助你快速地操作和推送创建好的镜像服务。
+```Docker
+docker pull <regsitry_path> # 从私有或公共仓库拉取镜像
+docker images # 查看镜像
+docker ps # 列举当前正在运行的容器
+docker container logs <container_id> # 查看日志
+docker stop <container_id> #停止容器
+docker start <container_id> #启动stop的容器
+docker rm <container_id> #删除容器
+docker push <registry_path>[image_version] #推送镜像到远端仓库
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-11-05-Docker-Sharing.md:82]: 查看容器ip也常用

**Review by Realvincentyuan — COMMENTED** [review]
建议补充一些内容

**Comment by Tina0330** [comment]
已经根据建议修改了，袁老板  @Realvincentyuan

**Pull request merged**